src/components/icon/index.css

```diff
@@ -0,0 +1,6 @@
+.icon {
```

Comment:
When I run `npm run build:css` it generates file like this. The class name has suffix weird hash. Will be this hash different from build to build? What if I want to change the icon color in let's say Reader? 

```
.index-icon-3kEAc {
  display: inline-block;
  width: 1.5rem;
  height: 1.5rem;
  fill: red;
}
```

Reply:
If you go below, you fill is `currentColor`. This is a special keyword that means the fill inherits current element `color` property that is inherited from parent element.

Reply:
I meant it like it would be weird to use this selector `.index-icon-3kEAc` for changing any property of icon in Reader. Color was wrong example. Let's say I would want to change width.

Reply:
You pass another class name. It will be supported later. I am considering a way but I feel I do not have any other [option](https://github.com/JedWatson/classnames).

Reply:
To answer your question better, hash is base64-encoded selector content, so it's the same if style did not change and different otherwise. It's added to make a class name unique.

Reply:
If the only option is to pass another class name then I am strongly against this approach as I don't see any additional value. It will be just a big mess in imported packages. If we want to have own CSS framework we need to have some helper classes. For example `d-flex`. I won't be able to use it in for example in Reader unless I write something like `d-flex-dhdjf23` which is a mess.

Reply:
Please, take your time to read the [docs](https://github.com/css-modules/css-modules).

I've chosen CSS modules because I could not easily contribute to the Reader because of class name conflict. So, I decided that CSS file with ugly class names on production is not that bad.

Reply:
I was exactly like you 2 years ago, by the way 🙂

Reply:
I have not read the documentation yet. But it doesn't change my mind regarding unique names. We are building reusable library similar as bootstrap and it doesn't use base64 suffixes too. I can see use case with unique classnames in final product but not in reusable library. if the classnames conflicts are the reason then add just prefixes to all classes like `.deisgn-icon`

Reply:
It's not CSS framework. It's a component library. The difference is that you ideally do not have to care about CSS, just take a component and use it.

If you mind, we can switch CSS modules off. However, we will have mess when switch it on back. So I would no like to do it.

Reply:
I don't want to switch off css modules I just want you to remove suffixes. That's all. 

IMHO we should not design only components but also create helpers classes and layout too. We can borrow bootstrap or anything else for the start I don't mind. The idea is to have as much as possible of css code in this library and minumum css code in packages that will use this library.

Reply:
I agree but I am agains helpers kind of `.d-flex`. People often overuse it. You do not have to build components from helper classes. That is why I am against it.

However, I am not against helpers kind of `.h1`.

Reply:
> want you to remove suffixes

Removing suffixes means switching modules off. If we do so, we could not change the name of file, because it's used in final class name. Suffix if more about making class name weird to tell the user that he overrides it on his own risk because this might change.

It's basically taking ability to be the only one who can control class name. Others can only [extend the component but cannot change](https://en.wikipedia.org/wiki/Open–closed_principle).

Reply:
Your call then. The only thing I would want to see are helper classes without prefixes and suffixes at least.